client/allocrunner/taskrunner/service_hook.go

```diff
@@ -91,12 +95,17 @@ func (h *serviceHook) Poststart(ctx context.Context, req *interfaces.TaskPoststa
 	// Create task services struct with request's driver metadata
 	workloadServices := h.getWorkloadServices()
 
+	defer func() { h.initialRegistration = true }()
```

Comment:
This doesn't need to be in defer, specially given we don't care if RegisterWorkflow succeeds and `h.mu` is held for the entire duration of functions call including consul call.